lib/dialects/mysql/query.js

```diff
@@ -110,6 +110,8 @@ Query.prototype.formatResults = function(data) {
     result = data[0].version;
   } else if (this.isForeignKeysQuery()) {
     result = data;
+  } else if (this.isInsertQuery() || this.isUpdateQuery()) {
+    result = [result, data.affectedRows];
```

Comment:
an insert query would have affectedRows equal to insert values or a unique constraint, would it not?